external/fv3fit/fv3fit/_shared/pytorch/graphPredict.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import xarray as xr
+from fv3fit._shared import (
+    Predictor,
+    match_prediction_to_input_coords,
+    SAMPLE_DIM_NAME,
+    stack,
+)
+from typing import Any, Dict, Hashable, Iterable, Sequence
+
+
+class PytorchModel(Predictor):
+
+    _MODEL_FILENAME = "weight.pt"
+    _CONFIG_FILENAME = "config.yaml"
+    custom_objects: Dict[str, Any] = {}
+
+    def __init__(
+        self,
+        input_variables: Iterable[Hashable],
+        output_variables: Iterable[Hashable],
+        model: nn.Module,
+        unstacked_dims: Sequence[str],
+    ):
+        """Initialize the predictor
+        Args:
+            input_variables: names of input variables
+            output_variables: names of output variables
+            model: pytorch model to wrap
+            unstacked_dims: non-sample dimensions of model output
+        """
+        super().__init__(input_variables, output_variables)
+        self.input_variables = input_variables
+        self.output_variables = output_variables
+        self.model = model
+        self._unstacked_dims = unstacked_dims
+
+    def _array_prediction_to_dataset(
+        self, names, outputs, stacked_coords
+    ) -> xr.Dataset:
+        ds = xr.Dataset()
+        for name, output in zip(names, outputs):
+            dims = [SAMPLE_DIM_NAME] + list(self._unstacked_dims)
+            scalar_singleton_dim = (
+                len(output.shape) == len(dims) and output.shape[-1] == 1
+            )
+            if scalar_singleton_dim:  # remove singleton dimension
+                output = output[..., 0]
+                dims = dims[:-1]
+            da = xr.DataArray(
+                data=output, dims=dims, coords={SAMPLE_DIM_NAME: stacked_coords},
+            ).unstack(SAMPLE_DIM_NAME)
+            dim_order = [dim for dim in self._unstacked_dims if dim in da.dims]
+            ds[name] = da.transpose(*dim_order, ...)
+        return ds
+
+    def predict(self, X: xr.Dataset) -> xr.Dataset:
+        """Predict an output xarray dataset from an input xarray dataset."""
+        X = X.transpose("grid", "z")
```

Comment:
This might be OK for this PR, but you should really expect that the input dataset has dimensions `"tile", "x", "y", "z"`.